How Tor Works article (#1219)
privacyguides/privacyguides.org: docs/how-tor-works.md

---
title: "How does Tor work?"
icon: 'pg/tor'
---

The Tor network is an anonymity system designed to protect the privacy and anonymity of its users. Unlike a VPN service, Tor is both free to use and decentralized. below we will explain how it works.

## The Path Building

> Tor works by sending your traffic over a network of thousands of voluntarily run nodes (sometimes referred to as relays). Each node is a server that is run by volunteers to help you improve your privacy and anonymity. Every time you connect to Tor, it will choose three nodes to build a path to the internet; this is called a circuit. Each of these nodes has its own function:

Reviewer: These changes improve the overall flow of the sentences.

> Each node is a server that is run by volunteers to help you improve your privacy and anonymity.

This sentence is redundant.

I split up the sentences as it helps visually separate the different ideas. Idea 1 being how Tor works and it's decentralized nature. Idea 2 being what happens when you connect to Tor and the transition to the node descriptions.

```suggestion
Tor works by routing your traffic through a network comprised of thousands of voluntarily run servers called nodes (or relays).

Every time you connect to Tor, it will choose three nodes to build a path to the internet—this path is called a circuit. Each of these nodes has its own function:
```

>   - **The Entry Node**: often called the guard node, this is the first node your computer connects to. The entry node sees your IP address, but does not see what you are connecting to. Unlike the other nodes, the Tor client will randomly select an entry node, and stick with it for 2 to 3 months to protect you from certain attacks.

Reviewer: ```suggestion
  - **The Entry Node**: Often called the guard node, this is the first node your computer connects to. The entry node sees your IP address, but does not see what you are connecting to. Unlike the other nodes, the Tor client will randomly select an entry node, and stick with it for 2 to 3 months to protect you from certain attacks.
```

>   - **The Middle Node**: the second node to which your Tor client connects. This node can see which node traffic came from (the entry node) and which it goes to next. It does not, however, see your IP address, or the domain you are connecting to. This node is randomly picked from all Tor nodes for each circuit.

Reviewer: ```suggestion
  - **The Middle Node**: The second node to which your Tor client connects. This node can see which node traffic came from (the entry node) and which it goes to next. It does not, however, see your IP address, or the domain you are connecting to. This node is randomly picked from all Tor nodes for each circuit.
```

>   - **The Exit Node**: is where your traffic leaves the Tor network and is forwarded to the destination domain. The exit node does not know your IP (who you are) but it knows what you are connecting to. The exit node will, like the middle node, be chosen at random from the Tor nodes(if it runs with an exit flag).

Reviewer: Some more info on what a "exit flag" is might be useful.

```suggestion
  - **The Exit Node**: This is where your traffic leaves the Tor network and is forwarded to your desired destination. The exit node does not know your IP (who you are) but it knows what you are connecting to. The exit node will, like the middle node, be chosen at random from the Tor nodes (if it runs with an exit flag).
```

```mermaid

```

> The Encryption Tor will encrypt each packet three times, with each key in turn from the exit, middle and entry node in that order. Once Tor has built a circuit, browsing is done as follows:

Reviewer: > The Encryption Tor will encrypt

I'm not 100% sure what this is trying to say, but I took a shot at making it clearer.

It might also be helpful to include a popup on what a packet is. I feel like most people don't know what a packet is by name, but maybe I'm wrong.

```suggestion
The encryption method Tor uses will encrypt each packet three times. The encryption keys are derived from the exit, middle, and entry node—in that order.

Once Tor has built a circuit, browsing is done as follows:
```

> 1. When the packet arrives at the entry node the first layer of encryption is removed. In this encrypted packet it will find another encrypted packet with the middle node’s address. The entry node will then forward that to the middle node.

Reviewer: ```suggestion
1. When the packet arrives at the entry node, the first layer of encryption is removed. The entry node will see another encrypted packet with the middle node’s address, and, in turn, will forward it to the middle node.
```

> 2. When the middle node receives the packet from the entry node, it too will remove a layer of encryption with its key, and find an encrypted packet with the exit nodes address. The middle node will then forward the packet to exit node.

Reviewer: ```suggestion
2. When the middle node receives the packet from the entry node, it too will remove a layer of encryption with its key. The middle node will then see an encrypted packet with the exit node's address and forward it to the exit node.
```

> 3. When the exit node receives its packet, it will remove the last layer of encryption with its key, and find the destination address that the user wanted to connect to, and forward the packet to that address.

Reviewer: ```suggestion
3. When the exit node receives its packet, it will remove the last layer of encryption with its key. The exit node will see the destination address and forward the packet to that address.
```

> Here is an alternative visualization of the process. Note how each node removes its own layer of encryption, and when the destination website returns data, the same process happens entirely in reverse. For example, the exit node does not know who you are, but it DOES know which node it came from, so it adds his own layer of encryption, and sends it back.

Reviewer: ```suggestion
Here is an alternative visualization of the process. Note how each node removes its own layer of encryption, and when the destination website returns data, the same process happens entirely in reverse. For example, the exit node does not know who you are, but it does know which node it came from, and so it adds its own layer of encryption and sends it back.
```

```mermaid

```

> So what do we learn from this? Well we learn that Tor allows us to connect to a website without any single party knowing the entire path. The entry node knows who you are, but not where you are going; the middle node doesn’t know who you are OR where you are going; and the exit node knows where you are going, but not who you are. Because the exit node makes the connection, the destination website will never know who you are (the IP address of the originating device).

Reviewer: ```suggestion
So, what do we learn from this? We learn that Tor allows us to connect to a website without any single party knowing the entire path. The entry node knows who you are, but not where you are going; the middle node doesn’t know who you are or where you are going; and the exit node knows where you are going, but not who you are. Because the exit node makes the connection, the destination website will never know who you are (your IP address).
```
